lepton/qemu.go

```diff
@@ -228,6 +229,22 @@ func (q *qemu) addOption(flag, value string) {
 	q.flags = append(q.flags, option)
 }
 
+func (q *qemu) addAccel() {
+	// hv_support should not throw errors. When error/log handling is
+	// implemented, then errors in detecting hardware acceleration support
+	// should be reported. Here we treat any error in detecting hardware
+	// acceleration the same as detecting no hardware acceleration - we don't
+	// add the flag.
+	if ok, err := hv_support(); ok && err == nil {
+		if runtime.GOOS == "darwin" {
+			q.addOption("-accel", "hvf")
+			q.addOption("-cpu", "host")
+		} else {
+			q.addOption("-accel", "kvm")
```

Comment:
which qemu is this? on both qemu 3.x on mac and 2.5 on ubuntu I don't see accel flags

Reply:
As far as I can tell from the change logs the [-accel flag is 3.0+](https://wiki.qemu.org/ChangeLog/3.0). On qemu 3.1.0 on `macos` I see the flags listed on the man page. 

There can be a version check to choose the correct option/flag to avoid any deprecation warnings.

Reply:
Is should be noted that the default `qemu` from apt on `debian 9 stretch` is 2.8.

Reply:
`-accel` seems to be 2.12+ not 3.0+, earlier change logs do not mention.

Reply:
accel=kvm and -enable-kvm are equivalent,  just add `-enable-kvm` and remove the linux version check since `-enable-kvm` works across all versions.

Reply:
@tijoytom Will do, I have been trying to run things with `kvm` enabled (by any option) with no luck. The command hangs. Digging through mailing lists and forums reveals some bugs that inidicate that the "-accel kvm/-enable-kvm" options can hang and yet none of the proposed solutions worked for me. I've tried different kernels and different distros and still no luck. 

Would be you able to confirm this by `ops run -v <binary>` and they by sticking an `-enable-kvm` at the end of the command?

Reply:
i was going to respond to this - i'm pretty sure you have to set the acclerator used on osx ->

```
-accel hax
HAX is working and emulator runs in fast virt mode.
qemu-system-x86_64: terminating on signal 2 from pid 24570 (<unknown process>)
```

since kvm is linux specific it makes sense you have to set the appropriate flag here

Reply:
what's  your qemu version?

Reply:
I see the same problem, i think it might have to do something with qemu version, i know for sure 2.5 worked. I dont have qemu2.5 now, if you have it ready can you please try it.

Reply:
I've tried on `2.8.1` which is the default on `Debian 9 stretch` and I have tried on `3.1.0` which is the default on `Manjaro` (I know manjaro is not supported I just tried it anyway since I have it available). I have also tried it on `3.1.0` on `Debian 9` and `3.1.0` on `Fedora29`.

Reply:
Actually that is `3.0.0` on `fedora29`.

Reply:
@tijoytom You were correct. `-enable-kvm`  works fine on `2.5.0`. In fact it works fine on `2.5.0`, `2.6.0`... all the way to `2.8.0` where the ops generated command stops working.

For whatever reason, after qemu-version `2.7.1`, you must disable hardware `irq` handling and allow qemu to continue to fully emulate `irq` when using `-enable-kvm`.

So AFACT:

**When using the flag `-enable-kvm` we must also use `-machine kernel-irqchip=off`**

Unless there is some other configuration which allows kvm with kernel-irqchip that I am not seeing. The qemu docs aren't the greatest in this area.

Reply:
need to let @wjhun && @mkhon chime in here - I think this is the msi vs. msi-x problem you're hitting https://github.com/nanovms/nanos/pull/172 && https://github.com/nanovms/nanos/issues/27 - kernel-irqchip=off prob. is not the fix we want to do here

Reply:
I see.

Reply:
I wonder what the status of this is because with most of the default supported versions of `qemu` (those that are installed by default on supported distros) if a user chooses the `--bridged` option it enables `kvm` and qemu will hang. The `-kernel-irqchip=off` option may not be the optimal solution, but it does at least work and basic `time` benchmarks show that the image is executing significantly faster than without `--enable-kvm`.

Reply:
another option to look at (cause we've already reported this problem upstream) is this ->

I haven't personally looked at the results between your solution and this but it'd be interesting to get some stats on it

```
-global driver=virtio-blk-pci,property=ioeventfd,value=off
```

the other thing is that I'm not opposed to turning this on selectively right now for the affected qemu versions as long as we let the others remain w/out it

lastly - we are in all likelihood converting to msi-x in the very near future which kinda obviates this entire issue - so this would be more of a thing if you want to spend time on it now or just wait